Stack/Opc.Ua.Core/Stack/State/DataTypeState.cs

```diff
@@ -221,6 +221,11 @@ public override void Update(ISystemContext context, BinaryDecoder decoder, Attri
                         value = dataTypeDefinition;
                     }
 
+                    if (value == null && result == null)
```

Comment:
is this by intention value == null && result == null, or should it read value == null || result == null?
While I do understand what the code does, I have difficulties to figure out what it should do.

Reply:
hi @ThomasNehring, in this case result is null which means good but then only if the value is also null its a bad attribute. Otherwise return the result. Safer may have been ServiceResult.IsGood(result) for the second term